Source/ACE.Server/WorldObjects/SkillAlterationDevice.cs

```diff
@@ -0,0 +1,250 @@
+
+                                break;
+                            }
+                        }
+                        else
+                        {
+                            player.Session.Network.EnqueueSend(new GameEventWeenieErrorWithString(player.Session, WeenieErrorWithString.NotEnoughSkillCreditsToSpecialize, currentSkill.Skill.ToSentence()));
+                            player.Session.Network.EnqueueSend(new GameEventUseDone(player.Session, WeenieError.YouFailToAlterSkill));
+                            break;
+                        }
+                    }
+
+                    //Tried to use a specialization gem on a skill that is either already specialized, or untrained
+                    player.Session.Network.EnqueueSend(new GameEventWeenieErrorWithString(player.Session, WeenieErrorWithString.Your_SkillMustBeTrained, currentSkill.Skill.ToSentence()));
+                    player.Session.Network.EnqueueSend(new GameEventUseDone(player.Session, WeenieError.YouFailToAlterSkill));
+                    break;
+                case SkillAlterationType.Lower:
+                    //We're using a Gem of Forgetfullness
+
+                    //Check for equipped items that have requirements in the skill we're lowering
+                    if (CheckWieldedItems(player))
+                    {
+                        //Items are wielded which might be affected by a lowering operation
+                        player.Session.Network.EnqueueSend(new GameEventWeenieErrorWithString(player.Session, WeenieErrorWithString.CannotLowerSkillWhileWieldingItem, currentSkill.Skill.ToSentence()));
+                        player.Session.Network.EnqueueSend(new GameEventUseDone(player.Session, WeenieError.YouFailToAlterSkill));
+                        break;
+                    }
+
+                    if (currentSkill.AdvancementClass == SkillAdvancementClass.Specialized)
+                    {
+                        if (player.UnspecializeSkill(currentSkill.Skill, currentSkillCost.SpecializationCost))
+                        {
+                            //Unspecialization was successful, notify the client
+                            player.Session.Network.EnqueueSend(new GameMessagePrivateUpdateSkill(player, currentSkill.Skill, SkillAdvancementClass.Trained, 0, 0, 0));
+                            player.Session.Network.EnqueueSend(new GameMessagePrivateUpdatePropertyInt(player, PropertyInt.AvailableSkillCredits, player.AvailableSkillCredits ?? 0));
+                            player.Session.Network.EnqueueSend(new GameEventWeenieErrorWithString(player.Session, WeenieErrorWithString.YouHaveSucceededUnspecializing_Skill, currentSkill.Skill.ToSentence()));
+                            player.Session.Network.EnqueueSend(new GameEventUseDone(player.Session, WeenieError.None));
+
+                            //Destroy the gem we used successfully
+                            Destroy();
```

Comment:
Should be using player.TryRemoveItemFromInventoryWithNetworking(this, 1) without the line following the Destroy()

Reply:
Fixed